File: testData/src/java8/pkg/TestTryLoopRecompile.java
package pkg;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class TestTryLoopRecompile {
  private boolean field;

  public void test(File file) {
    while(true) {
      try {
        if (this.field) {
          new Scanner(file);
          continue;
        }
      } catch (FileNotFoundException var3) {
        var3.printStackTrace();
      }

      return;
    }
  }
}

File: plugins/kotlin/testData/src/kt/pkg/TestTryLoopRecompile.kt
package pkg

import java.io.File
import java.io.FileNotFoundException
import java.util.*

class TestTryLoopRecompile {
    private val field = false

    fun test(file: File) {
        while (true) {
            try {
                if (this.field) {
                    Scanner(file)
                    continue
                }
            } catch (var3: FileNotFoundException) {
                var3.printStackTrace()
            }

            return
        }
    }
}
